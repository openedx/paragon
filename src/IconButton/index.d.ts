import * as React from 'react';

export type IconButtonVariants = 'primary' | 'secondary' | 'success' | 'warning' | 'danger' | 'light' | 'dark' | 'black' | 'brand';
export type IconButtonSizes = 'sm' | 'md' | 'inline';

export interface IconButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  className?: string;
  iconAs?: keyof JSX.IntrinsicElements;
  src?: React.ReactElement | React.FC;
  alt: string;
  invertColors?: boolean;
  icon?: {
    prefix: string;
    iconName: string;
    icon: any[];
  };
  iconClassNames?: string;
  onClick?: (event: React.MouseEvent<HTMLButtonElement>) => void;
  variant?: IconButtonVariants;
  size?: IconButtonSizes;
  isActive?: boolean;
}

export interface IconButtonWithTooltipProps extends Omit<IconButtonProps, 'onClick'> {
  tooltipPlacement?: string;
  tooltipContent: React.ReactNode;
}

declare const IconButtonWithTooltip: React.ForwardRefExoticComponent<
IconButtonWithTooltipProps & React.RefAttributes<HTMLButtonElement>>;

declare const IconButton: React.ForwardRefExoticComponent<
IconButtonProps & React.RefAttributes<HTMLButtonElement>> & {
  IconButtonWithTooltip: typeof IconButtonWithTooltip;
};

export default IconButton;
export { IconButtonWithTooltip };
