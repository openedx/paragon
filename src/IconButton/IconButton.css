/**
 * Do not edit directly
 * Generated on Wed, 01 Mar 2023 00:27:20 GMT
 */

:root {
  --pgn-size-icon-button-diameter-sm: 2.25rem;
  --pgn-size-icon-button-diameter-md: 2.75rem;
  --pgn-size-icon-button-diameter-inline: calc(var(--pgn-typography-line-height-base) + .1em);
}
