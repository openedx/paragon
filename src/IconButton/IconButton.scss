$icon-button-colors: (
  "primary": $primary,
  "secondary": $secondary,
  "brand": $brand,
  "success": $success,
  "warning": $warning,
  "danger": $danger,
  "light": $light,
  "dark": $dark,
  "black": $black,
) !default;
$btn-icon-diameter-md: 2.75rem !default;
$btn-icon-diameter-sm: 2.25rem !default;
$btn-icon-diameter-inline: calc(#{$line-height-base}em + .1em) !default;

$btn-icon-bg: transparent !default;
$btn-icon-accent-color: white !default;

@mixin btn-icon-size($icon-size) {
  width: $icon-size;
  height: $icon-size;
}

@mixin btn-icon-variant($icon-color, $accent-color, $active-state: false) {
  &.hover,
  &:hover,
  &.active,
  &:active {
    background-color: $icon-color;
    color: $accent-color;
  }

  @if $active-state {
    background-color: $icon-color;
    color: $accent-color;

    &.focus,
    &:focus {
      outline: 2px transparent;
      background-color: $icon-color;
      color: $accent-color;
    }
  }

  @else {
    background-color: $btn-icon-bg;
    color: $icon-color;

    &.focus,
    &:focus {
      outline: 2px transparent;
      background-color: $btn-icon-bg;
      color: $icon-color;
      box-shadow: inset 0 0 0 $btn-focus-width $icon-color;
    }
  }
}

.btn-icon {
  @include btn-icon-size($btn-icon-diameter-md);

  border-radius: 50%;
  padding: 0;
  display: inline-flex;
  justify-content: center;
  border: none;

  &.btn-icon-sm {
    @include btn-icon-size($btn-icon-diameter-sm);
  }

  &.btn-icon-inline {
    @include btn-icon-size($btn-icon-diameter-inline);

    padding: 1em;
    vertical-align: middle;

    .btn-icon__icon-container {
      height: 100%;
      width: 100%;
      justify-content: center;
      align-items: center;
    }
  }

  .btn-icon__icon-container {
    display: inline-flex;
    align-self: center;

    .pgn__icon.btn-icon__icon {
      height: 1.5rem;
      width: 1.5rem;
    }
  }

  .btn-icon__icon {
    height: 1em;
    width: 1em;
  }

  &.disabled,
  &:disabled {
    opacity: $btn-disabled-opacity;
    pointer-events: none;
  }

  @each $name, $color in $icon-button-colors {
    &.btn-icon-#{$name} {
      @include btn-icon-variant($color, $btn-icon-accent-color);
    }

    &.btn-icon-inverse-#{$name} {
      @include btn-icon-variant($btn-icon-accent-color, $color);
    }

    &.btn-icon-#{$name}-active {
      @include btn-icon-variant($color, $btn-icon-accent-color, true);
    }

    &.btn-icon-inverse-#{$name}-active {
      @include btn-icon-variant($btn-icon-accent-color, $color, true);
    }
  }
}
