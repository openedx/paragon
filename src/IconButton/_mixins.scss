@mixin btn-icon-size($icon-size) {
  width: $icon-size;
  height: $icon-size;
}
