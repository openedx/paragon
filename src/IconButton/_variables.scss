$icon-button-colors: (
  "primary": $primary,
  "secondary": $secondary,
  "brand": $brand,
  "success": $success,
  "warning": $warning,
  "danger": $danger,
  "light": $light,
  "dark": $dark,
  "black": $black,
) !default;
$btn-icon-diameter-md: 2.75rem !default;
$btn-icon-diameter-sm: 2.25rem !default;
$btn-icon-diameter-inline: calc(#{$line-height-base}em + .1em) !default;

$btn-icon-bg: transparent !default;
$btn-icon-accent-color: white !default;
