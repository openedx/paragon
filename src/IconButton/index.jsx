import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { OverlayTrigger } from '../Overlay';
import Tooltip from '../Tooltip';

const IconButton = React.forwardRef(({
  className,
  alt,
  invertColors,
  icon,
  src,
  iconClassNames,
  onClick,
  size,
  variant,
  iconAs,
  isActive,
  ...attrs
}, ref) => {
  const invert = invertColors ? 'inverse-' : '';
  const activeStyle = isActive ? `${variant}-` : '';
  if (!iconAs && process.env.NODE_ENV === 'development' && console) {
    const msg = '[Deprecated] IconButton: you have not provided a value for iconAs prop and '
      + 'are using a default one - FontAwesomeIcon, the default value is going to be changed soon '
      + 'as Paragon is moving away from FontAwesome, please use Paragon\'s icons instead.';
    // eslint-disable-next-line no-console
    console.warn(msg);
  }
  const IconComponent = iconAs || FontAwesomeIcon;
  return (
    <button
      aria-label={alt}
      className={classNames(
        'btn-icon',
        `btn-icon-${invert}${variant}`,
        `btn-icon-${size}`,
        {
          [`btn-icon-${invert}${activeStyle}active`]: isActive,
        },
        className,
      )}
      onClick={onClick}
      type="button"
      ref={ref}
      {...attrs}
    >
      <span className="btn-icon__icon-container">
        <IconComponent
          className={classNames('btn-icon__icon', iconClassNames)}
          icon={icon}
          src={src}
        />
      </span>
    </button>
  );
});

IconButton.defaultProps = {
  iconAs: undefined,
  src: null,
  icon: undefined,
  iconClassNames: undefined,
  className: undefined,
  invertColors: false,
  variant: 'primary',
  size: 'md',
  onClick: () => {},
  isActive: false,
};

IconButton.propTypes = {
  /** A custom class name. */
  className: PropTypes.string,
  /** Component that renders the icon, currently defaults to `FontAwesomeIcon`,
   *  but is going to be deprecated soon, please use Paragon's icons instead. */
  iconAs: PropTypes.elementType,
  /** An icon component to render. Example import of a Paragon icon component:
   * `import { Check } from '@edx/paragon/dist/icon';`
   * */
  src: PropTypes.oneOfType([PropTypes.element, PropTypes.func]),
  /** Alt text for your icon. For best practice, avoid using alt text to describe
   * the image in the `IconButton`. Instead, we recommend describing the function
   * of the button. */
  alt: PropTypes.string.isRequired,
  /** Changes icon styles for dark background */
  invertColors: PropTypes.bool,
  /** Accepts a React fontawesome icon. https://fontawesome.com/how-to-use/on-the-web/using-with/react */
  icon: PropTypes.shape({
    prefix: PropTypes.string,
    iconName: PropTypes.string,
    // eslint-disable-next-line react/forbid-prop-types
    icon: PropTypes.array,
  }),
  /** Extra class names that will be added to the icon */
  iconClassNames: PropTypes.string,
  /** Click handler for the button */
  onClick: PropTypes.func,
  /** Type of button (uses Bootstrap options) */
  variant: PropTypes.oneOf(['primary', 'secondary', 'success', 'warning', 'danger', 'light', 'dark', 'black', 'brand']),
  /** size of button to render */
  size: PropTypes.oneOf(['sm', 'md', 'inline']),
  /** whether to show the `IconButton` in an active state, whose styling is distinct from default state */
  isActive: PropTypes.bool,
};

/**
 *
 * @param { object } args Arguments
 * @param { string } args.tooltipPlacement choose from https://popper.js.org/docs/v2/constructors/#options
 * @param { React.Component } args.tooltipContent any content to pass to tooltip content area
 * @returns { IconButton } a button wrapped in overlaytrigger
 */
function IconButtonWithTooltip({
  tooltipPlacement, tooltipContent, variant, invertColors, ...props
}) {
  const invert = invertColors ? 'inverse-' : '';
  return (
    <OverlayTrigger
      placement={tooltipPlacement}
      overlay={(
        <Tooltip
          id={`iconbutton-tooltip-${tooltipPlacement}`}
          variant={invert ? 'light' : ''}
        >
          {tooltipContent}
        </Tooltip>
      )}
    >
      <IconButton variant={variant} invertColors={invertColors} {...props} />
    </OverlayTrigger>
  );
}

IconButtonWithTooltip.defaultProps = {
  tooltipPlacement: 'top',
  variant: 'primary',
  invertColors: false,
};

IconButtonWithTooltip.propTypes = {
  /** tooltip placement can be top, left, right etc, per https://popper.js.org/docs/v2/constructors/#options  */
  tooltipPlacement: PropTypes.string,
  /** any valid JSX or text to be rendered as tooltip contents */
  tooltipContent: PropTypes.node.isRequired,
  /** Type of button (uses Bootstrap options) */
  variant: PropTypes.oneOf(['primary', 'secondary', 'success', 'warning', 'danger', 'light', 'dark', 'black', 'brand']),
  /** Changes icon styles for dark background */
  invertColors: PropTypes.bool,
};

IconButton.IconButtonWithTooltip = IconButtonWithTooltip;

export default IconButton;
export { IconButtonWithTooltip };
