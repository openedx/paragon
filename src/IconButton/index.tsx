import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { type Placement } from 'react-bootstrap/Overlay';

import { OverlayTrigger } from '../Overlay';
import Tooltip from '../Tooltip';
import Icon from '../Icon';

interface Props extends React.HTMLAttributes<HTMLButtonElement> {
  iconAs?: typeof Icon | typeof FontAwesomeIcon,
  /** Additional CSS class[es] to apply to this button */
  className?: string;
  /** Alt text for your icon. For best practice, avoid using alt text to describe
   * the image in the `IconButton`. Instead, we recommend describing the function
   * of the button. */
  alt: string;
  /** Changes icon styles for dark background */
  invertColors?: boolean;
  /** An icon component to render. Example import of a Paragon icon component:
   * `import { Check } from '@openedx/paragon/icons';`
   * */
  // Note: React.ComponentType is what we want here. React.ElementType would allow some element type strings like "div",
  // but we only want to allow components like 'Add' (a specific icon component function/class)
  src?: React.ComponentType;
  /** Extra class names that will be added to the icon */
  iconClassNames?: string;
  /** Click handler for the button */
  onClick?: React.MouseEventHandler<HTMLButtonElement>;
  /** whether to show the `IconButton` in an active state, whose styling is distinct from default state */
  isActive?: boolean;
  /** @deprecated Using FontAwesome icons is deprecated. Instead, pass iconAs={Icon} src={...} */
  icon?: { prefix?: string; iconName?: string, icon?: any[] },
  /** Type of button (uses Bootstrap options) */
  variant?: 'primary' | 'secondary' | 'success' | 'warning' | 'danger' | 'light' | 'dark' | 'black' | 'brand';
  /** size of button to render */
  size?: 'sm' | 'md' | 'inline';
  /** Used with `IconButtonToggle` */
  value?: string;
  /** no children */
  children?: never;
}

const IconButton = React.forwardRef<HTMLButtonElement, Props>(({
  className,
  alt,
  invertColors,
  icon,
  src,
  iconClassNames,
  onClick,
  size,
  variant,
  iconAs,
  isActive,
  children, // unused, just here because we don't want it to be part of 'attrs'
  ...attrs
}, ref) => {
  const invert = invertColors ? 'inverse-' : '';
  const activeStyle = isActive ? `${variant}-` : '';
  if (!iconAs && process.env.NODE_ENV === 'development' && console) {
    const msg = '[Deprecated] IconButton: you have not provided a value for iconAs prop and '
      + 'are using a default one - FontAwesomeIcon, the default value is going to be changed soon '
      + 'as Paragon is moving away from FontAwesome, please use Paragon\'s icons instead.';
    // eslint-disable-next-line no-console
    console.warn(msg);
  }
  const IconComponent = iconAs || FontAwesomeIcon;
  return (
    <button
      aria-label={alt}
      className={classNames(
        'btn-icon',
        `btn-icon-${invert}${variant}`,
        `btn-icon-${size}`,
        {
          [`btn-icon-${invert}${activeStyle}active`]: isActive,
        },
        className,
      )}
      onClick={onClick}
      type="button"
      ref={ref}
      {...attrs}
    >
      <span className="btn-icon__icon-container">
        <IconComponent
          className={classNames('btn-icon__icon', iconClassNames)}
          icon={icon as any}
          src={src}
        />
      </span>
    </button>
  );
});

IconButton.defaultProps = {
  iconAs: undefined,
  src: undefined,
  icon: undefined,
  iconClassNames: undefined,
  className: undefined,
  invertColors: false,
  variant: 'primary',
  size: 'md',
  onClick: () => {},
  isActive: false,
  value: undefined,
  children: undefined,
};

IconButton.propTypes = {
  /** A custom class name. */
  className: PropTypes.string,
  /** Component that renders the icon, currently defaults to `FontAwesomeIcon`,
   *  but is going to be deprecated soon, please use Paragon's icons instead. */
  iconAs: PropTypes.elementType as any,
  /** An icon component to render. Example import of a Paragon icon component:
   * `import { Check } from '@openedx/paragon/icons';`
   * */
  src: PropTypes.elementType as any,
  /** Alt text for your icon. For best practice, avoid using alt text to describe
   * the image in the `IconButton`. Instead, we recommend describing the function
   * of the button. */
  alt: PropTypes.string.isRequired,
  /** Changes icon styles for dark background */
  invertColors: PropTypes.bool,
  /** Accepts a React fontawesome icon. */
  icon: PropTypes.shape({
    prefix: PropTypes.string,
    iconName: PropTypes.string,
    // eslint-disable-next-line react/forbid-prop-types
    icon: PropTypes.array,
  }) as any,
  /** Extra class names that will be added to the icon */
  iconClassNames: PropTypes.string,
  /** Click handler for the button */
  onClick: PropTypes.func,
  /** Type of button (uses Bootstrap options) */
  variant: PropTypes.oneOf(['primary', 'secondary', 'success', 'warning', 'danger', 'light', 'dark', 'black', 'brand']),
  /** size of button to render */
  size: PropTypes.oneOf(['sm', 'md', 'inline']),
  /** whether to show the `IconButton` in an active state, whose styling is distinct from default state */
  isActive: PropTypes.bool,
  /** Used with <IconButtonToggle> */
  value: PropTypes.string,
};

interface PropsWithTooltip extends Props {
  /** choose from https://popper.js.org/docs/v2/constructors/#options */
  tooltipPlacement: Placement,
  /** any content to pass to tooltip content area */
  tooltipContent: React.ReactNode,
}

/**
 * An icon button wrapped in overlaytrigger to display a tooltip.
 */
function IconButtonWithTooltip({
  tooltipPlacement, tooltipContent, ...props
}: PropsWithTooltip) {
  const invert = props.invertColors ? 'inverse-' : '';
  return (
    <OverlayTrigger
      placement={tooltipPlacement}
      overlay={(
        <Tooltip
          id={`iconbutton-tooltip-${tooltipPlacement}`}
          variant={invert ? 'light' : undefined}
        >
          {tooltipContent}
        </Tooltip>
      )}
    >
      <IconButton {...props} />
    </OverlayTrigger>
  );
}

IconButtonWithTooltip.defaultProps = {
  ...IconButton.defaultProps,
  tooltipPlacement: 'top',
};

IconButtonWithTooltip.propTypes = {
  /** tooltip placement can be top, left, right etc, per https://popper.js.org/docs/v2/constructors/#options  */
  tooltipPlacement: PropTypes.string,
  /** any valid JSX or text to be rendered as tooltip contents */
  tooltipContent: PropTypes.node.isRequired,
  /** Type of button (uses Bootstrap options) */
  variant: PropTypes.oneOf(['primary', 'secondary', 'success', 'warning', 'danger', 'light', 'dark', 'black', 'brand']),
  /** Changes icon styles for dark background */
  invertColors: PropTypes.bool,
};

(IconButton as any).IconButtonWithTooltip = IconButtonWithTooltip;

export default IconButton as typeof IconButton & {
  IconButtonWithTooltip: typeof IconButtonWithTooltip,
};
export { IconButtonWithTooltip };
