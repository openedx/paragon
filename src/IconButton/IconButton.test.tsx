import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';

import { InfoOutline } from '../../icons';
import IconButton from '.';
import Icon from '../Icon';

describe('<IconButton />', () => {
  const alt = 'alternative';
  const iconAs = Icon;
  const src = InfoOutline;
  const variant = 'secondary' as const;
  const props = {
    alt,
    src,
    iconAs,
    variant,
  };
  const deprecatedFontAwesomeExample = {
    prefix: 'pgn',
    iconName: 'copy',
    icon: [
      448,
      512,
      [],
      'f0c5',
      'M384 336l-192 0c-8.8 0-16-7.2-16-16l0-256c0-8.8 7.2-16 16-16l140.1 0L400 115.9 400 320c0 8.8-7.2 16-16 16zM192 384l192 0c35.3 0 64-28.7 64-64l0-204.1c0-12.7-5.1-24.9-14.1-33.9L366.1 14.1c-9-9-21.2-14.1-33.9-14.1L192 0c-35.3 0-64 28.7-64 64l0 256c0 35.3 28.7 64 64 64zM64 128c-35.3 0-64 28.7-64 64L0 448c0 35.3 28.7 64 64 64l192 0c35.3 0 64-28.7 64-64l0-32-48 0 0 32c0 8.8-7.2 16-16 16L64 464c-8.8 0-16-7.2-16-16l0-256c0-8.8 7.2-16 16-16l32 0 0-48-32 0z',
    ],
  };
  it('renders with required props', () => {
    const tree = renderer.create((
      <IconButton iconAs={Icon} src={InfoOutline} alt={alt} />
    )).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders with deprecated props', () => {
    const tree = renderer.create((
      <IconButton icon={deprecatedFontAwesomeExample} alt={alt} />
    )).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('passes the alt text to the button aria-label', () => {
    const { getByRole } = render(<IconButton {...props} onClick={() => {}} />);
    const button = getByRole('button');
    expect(button).toHaveAttribute('aria-label', alt);
  });
  it('should render with active style if isActive is true', () => {
    const { container } = render(<IconButton {...props} isActive onClick={() => {}} />);
    const button = container.querySelector(`.btn-icon-${variant}-active`);
    expect(button).toBeInTheDocument();
  });
  it('should render with inverse active style if inverse and isActive is true', () => {
    const { container } = render(<IconButton {...props} isActive invertColors onClick={() => {}} />);
    const button = container.querySelector(`.btn-icon-inverse-${variant}-active`);
    expect(button).toBeInTheDocument();
  });
  it('should not render with inverse- class names if invertColors is false', () => {
    const { container } = render(<IconButton {...props} onClick={() => {}} />);
    const inverseButton = container.querySelector(`.btn-icon-inverse-${variant}`);
    const inverseActiveButton = container.querySelector(`.btn-icon-inverse-${variant}-active`);
    expect(inverseButton).not.toBeInTheDocument();
    expect(inverseActiveButton).not.toBeInTheDocument();
  });
  it('should render with inverse- class names if invertColors is true', () => {
    const { container } = render(<IconButton {...props} onClick={() => {}} invertColors />);
    const inverseButton = container.querySelector(`.btn-icon-inverse-${variant}`);
    expect(inverseButton).toBeInTheDocument();
  });
  it('should add the icon class names if it receives them', () => {
    const { container } = render(<IconButton {...props} onClick={() => {}} iconClassNames="foo bar" />);
    const iconWithFooClass = container.querySelector('.foo');
    const iconWithBarClass = container.querySelector('.bar');
    expect(iconWithFooClass).toBeInTheDocument();
    expect(iconWithBarClass).toBeInTheDocument();
  });

  describe('onClick', () => {
    it('performs the onClick action when clicked', async () => {
      const spy = jest.fn();
      const { getByRole } = render(<IconButton {...props} onClick={spy} />);
      const button = getByRole('button');
      await userEvent.click(button);
      expect(spy).toHaveBeenCalledTimes(1);
    });

    it('only clicks one icon at a time', async () => {
      const spy1 = jest.fn();
      const spy2 = jest.fn();
      const { getAllByRole } = render(
        <>
          <IconButton {...props} onClick={spy1} />
          <IconButton {...props} onClick={spy2} />
        </>,
      );

      const buttons = getAllByRole('button');

      await userEvent.click(buttons[0]);
      expect(spy1).toHaveBeenCalledTimes(1);
      expect(spy2).toHaveBeenCalledTimes(0);

      await userEvent.click(buttons[1]);
      expect(spy1).toHaveBeenCalledTimes(1);
      expect(spy2).toHaveBeenCalledTimes(1);
    });
  });

  describe('<IconButton.IconButtonWithTooltip>', () => {
    it('renders with required props', () => {
      const tree = renderer.create((
        <IconButton.IconButtonWithTooltip
          iconAs={Icon}
          src={InfoOutline}
          alt={alt}
          tooltipContent="Hello"
          tooltipPlacement="left-end"
        />
      )).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
