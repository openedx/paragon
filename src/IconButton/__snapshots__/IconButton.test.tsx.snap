// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconButton /> <IconButton.IconButtonWithTooltip> renders with required props 1`] = `
<button
  aria-label="alternative"
  className="btn-icon btn-icon-primary btn-icon-md"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  type="button"
>
  <span
    className="btn-icon__icon-container"
  >
    <span
      className="pgn__icon btn-icon__icon"
    >
      <svg
        aria-hidden={true}
        fill="none"
        focusable={false}
        height={24}
        role="img"
        viewBox="0 0 24 24"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11 7h2v2h-2V7Zm0 4h2v6h-2v-6Zm1-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2Zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8Z"
          fill="currentColor"
        />
      </svg>
    </span>
  </span>
</button>
`;

exports[`<IconButton /> renders with deprecated props 1`] = `
<button
  aria-label="alternative"
  className="btn-icon btn-icon-primary btn-icon-md"
  onClick={[Function]}
  type="button"
>
  <span
    className="btn-icon__icon-container"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-InfoOutlineIcon btn-icon__icon"
      data-icon="InfoOutlineIcon"
      data-prefix="pgn"
      focusable="false"
      role="img"
      style={{}}
      viewBox="0 0 function SvgInfoOutline(props) {
  return /*#__PURE__*/React.createElement("svg", _extends({
    width: 24,
    height: 24,
    viewBox: "0 0 24 24",
    fill: "none",
    xmlns: "http://www.w3.org/2000/svg"
  }, props), /*#__PURE__*/React.createElement("path", {
    d: "M11 7h2v2h-2V7Zm0 4h2v6h-2v-6Zm1-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2Zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8Z",
    fill: "currentColor"
  }));
} undefined"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        fill="currentColor"
        style={{}}
      />
    </svg>
  </span>
</button>
`;

exports[`<IconButton /> renders with required props 1`] = `
<button
  aria-label="alternative"
  className="btn-icon btn-icon-primary btn-icon-md"
  onClick={[Function]}
  type="button"
>
  <span
    className="btn-icon__icon-container"
  >
    <span
      className="pgn__icon btn-icon__icon"
    >
      <svg
        aria-hidden={true}
        fill="none"
        focusable={false}
        height={24}
        role="img"
        viewBox="0 0 24 24"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11 7h2v2h-2V7Zm0 4h2v6h-2v-6Zm1-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2Zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8Z"
          fill="currentColor"
        />
      </svg>
    </span>
  </span>
</button>
`;
